client/app/bundles/course/assessment/question/programming/components/ProgrammingQuestionForm.jsx

```diff
@@ -0,0 +1,339 @@
+import React, { PropTypes } from 'react';
+import ReactSummernote from 'react-summernote';
+import Select from 'react-select';
+import { injectIntl, defineMessages } from 'react-intl';
+import styles from './ProgrammingQuestionForm.scss'
+
+import 'react-select/dist/react-select.css';
+
+const translations = defineMessages({
+  attemptLimitPlaceholderMessage: {
+    id: 'course.assessment.question.programming.programmingQuestionForm.attemptLimitPlaceholderMessage',
+    defaultMessage: 'The maximum times that the students can test their answers (does not apply to staff)',
+    description: 'Placeholder message for attempt limit input field.',
+  },
+  downloadPackage: {
+    id: 'course.assessment.question.programming.programmingQuestionForm.downloadPackage',
+    defaultMessage: 'Download Package',
+    description: 'Title for downloading the zip package.',
+  },
+  editTestsOnlineButton: {
+    id: 'course.assessment.question.programming.programmingQuestionForm.editTestsOnlineButton',
+    defaultMessage: 'Edit Tests Online',
```

Comment:
The text might not be fitting in the case where we create tests. Will "Use Test Case Editor" or just "Test Case Editor" be better?